### PR TITLE
FilterHelper: Rename filter functions

### PR DESCRIPTION
**Pull request by shreyans800755**
It renames `get_filtered_bears` to `apply_filter` and
`filter_bears` to `apply_filters`.

Closes https://github.com/coala/coala/issues/4586

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by Makman2** [comment]
ack b5906b0

**Comment by Makman2** [comment]
Would like to see a separate commit for the reordering of parameters^^

**Comment by shreyans800755** [comment]
Should I include renaming in the reordering commit as well ? Or separate commit for that too ?

**Comment by shreyans800755** [comment]
It also changes orders for `apply_filters` params.
And also renames params for both the functions to make it more meaningful.

File: coalib/parsing/FilterHelper.py [hunk 1]
```diff
@@ -21,34 +21,35 @@ def is_valid_filter(cls, filter):
...
-    def apply_filters(cls, args, bears=None):
+    def apply_filters(cls, filters, bears=None):
         """
         Returns bears after filtering based on ``args``. It returns
         intersection of bears if more than one element is present in ``args``
         list.
 
-        :param bears: The bears to filter.
-        :param args:  List of args based on ``bears`` has to be filtered.
+        :param filters:  List of args based on ``bears`` has to be filtered.
```
Comment (Makman2) [inline coalib/parsing/FilterHelper.py]: You have to reindent all parameters :/

There's actually another format which I try to make coala switch to, but our bears aren't ready yet to enforce that style:
```
description like before

even more description

:param xxx:
    4 space indent but on next line.
:return:
    stuff
```
If you like you could switch to it, but not necessary if you don't want it. However, often less lines have to be modified as indentation of params does not depend on the longest param name^^

**Comment by Makman2** [comment]
Please remove all `Closes xxx` references except on the commit changing the function names :)

File: coalib/parsing/FilterHelper.py [hunk 2]
```diff
@@ -21,34 +21,35 @@ def is_valid_filter(cls, filter):
...
                       For example:
                       ``[['language', 'c', 'java'], ['can_fix', 'syntax']]``
+        :param bears: The bears to filter.
```
Comment (Makman2) [inline coalib/parsing/FilterHelper.py]: The move of this `:param bears:` entry should go into the param-reordering commit :+1:

File: coalib/parsing/FilterHelper.py (hunk 1 above)
Comment (shreyans800755) [inline coalib/parsing/FilterHelper.py]: Should I keep indent in (-> side) on the next line of `For example` where actual list is mentioned ?

**Comment by Makman2** [comment]
ack 493831c 73d39ed b42d809

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**